### PR TITLE
generate debug log when exception occurred

### PR DESCRIPTION
**Pull request by syuu1228**
Using traceback_with_variables module, generate more detail traceback
with variables into debug log.
This will help fixing bugs which is hard to reproduce.

This is scylla-machine-image version of https://github.com/scylladb/scylla/pull/10472.
Please merge this only after https://github.com/scylladb/scylla/pull/10472 get merged, otherwise we will fail to load traceback_with_variables.

### CODE REVIEW

**Comment by syuu1228** [comment]
Fixed traceback error on unit test, rebased latest next, changed debug log directory to /var/tmp/scylla (aligned with scylla main repo https://github.com/scylladb/scylla/commit/7501465b7c9f91dad08245e5f7b9c7d1de86a490).

**Comment by syuu1228** [comment]
AMI test passed: https://jenkins.scylladb.com/job/scylla-master/job/releng-testing/job/ami/1419/

File: dist/redhat/build_rpm.sh [hunk 1]
```diff
@@ -65,10 +65,14 @@ pkg_install git
...
 pkg_install python3-pip
+pkg_install gcc
 
 echo "Running unit tests"
 cd tests
 sudo pip3 install pyyaml==5.3
+sudo pip3 install psutil
+sudo pip3 install distro
+sudo pip3 install traceback-with-variables
 python3 -m unittest test_scylla_configure.py
```
Comment (fruch) [inline dist/redhat/build_rpm.sh:76]: only the build_rpm.sh has the unittest running ? the build_deb.sh doesn't ?

Comment (fruch) [reply dist/redhat/build_rpm.sh:76]: the idea was to force scylla-pkg run them at build time. (before we using github actions, and before SMI has it's own next process)

also seem like it's not running the whole suite of tests... only test_scylla_configure.py.

maybe this section can be removed.

Comment (syuu1228) [reply dist/redhat/build_rpm.sh]: Dropped unit section from build_rpm.sh

**Comment by syuu1228** [comment]
Rebased with next.

**Comment by syuu1228** [comment]
Test passed: https://jenkins.scylladb.com/job/scylla-master/job/releng-testing/job/next-machine-image/399/

**Comment by syuu1228** [comment]
Test passed: https://jenkins.scylladb.com/job/scylla-master/job/releng-testing/job/next-machine-image/446/

**Comment by syuu1228** [comment]
@yaronkaikov ping

**Review by yaronkaikov — APPROVED** [review]

**Comment by yaronkaikov** [comment]
@fruch any other comments ? or can i merge it ?

**Comment by syuu1228** [comment]
Rebased with latest master.
@yaronkaikov @fruch ping, we want to have this to make debug scripts easier.

**Review by fruch — APPROVED** [review]
LGTM

**Pull request merged**